### PR TITLE
Setup remote deadline list storage with Firebase

### LINKED ISSUE #35: Implement support for firebase storage
Implement model representing the user's deadlines, so we can display them in the UI and store them somewhere (on local storage for now, ultimately on firebase).

Edit: the model can just be a simple `List<Deadline>` object, so I will try to implement support for firebase in the PR.

### PR DESCRIPTION
**Pull request by felko**
This PR implements remote data storage on Firebase to store the list of deadlines.
For now, the list of deadlines is not associated to any user, as we don't have authentication yet (waiting for #11), therefore the list is shared by all users.
We will integrate this feature to the corresponding UI view implemented by #36.

Closes #35.

### CODE REVIEW

File: app/src/main/java/com/github/multimatum_team/multimatum/repository/DeadlineRepository.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                task.result.filter { it.date > dateLimit }
+            } else {
+                listOf()
+            }
+        }
+
+    /**
+     * Add new deadline to the repository.
+     */
+    fun put(deadline: Deadline): Task<Unit>
```
Comment (Matathiasq13824) [inline app/src/main/java/com/github/multimatum_team/multimatum/repository/DeadlineRepository.kt:33]: It is normal to have this empty function ?

File: app/src/main/java/com/github/multimatum_team/multimatum/repository/FirebaseDeadlineRepository.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Remote Firebase repository for storing deadlines.
+ */
+class FirebaseDeadlineRepository : DeadlineRepository {
+    private var database: FirebaseDatabase = FirebaseDatabase.getInstance()
+
+    private val deadlinesRef
+        get() = database.getReference("deadlines")
+
+    override fun fetchAll(): Task<List<Deadline>> =
```
Comment (Matathiasq13824) [inline app/src/main/java/com/github/multimatum_team/multimatum/repository/FirebaseDeadlineRepository.kt:18]: Maybe add some commentaries to help understand what this code do

**Review by Matathiasq13824 — COMMENTED** [review]
I have seen that the tests hadn't worked, what is the problem ?

File: app/build.gradle [hunk 3]
```diff
@@ -63,6 +64,11 @@ dependencies {
     //implementation for the QRCode Generator
     implementation 'com.journeyapps:zxing-android-embedded:4.2.0'
     implementation 'com.google.zxing:core:3.4.0'
+
+    // Firebase dependencies
+    implementation platform('com.google.firebase:firebase-bom:29.2.0')
+    implementation 'com.google.firebase:firebase-database-ktx'
+    // implementation 'com.google.firebase:firebase-analytics-ktx'
```
Comment (macintosh1398) [inline app/build.gradle:71]: Did you forget to remove this comment ? (If not just ignore my comment).

File: build.gradle [hunk 4]
```diff
@@ -1,4 +1,13 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
+buildscript {
+    buildscript {
+        dependencies {
+            classpath 'com.google.gms:google-services:4.3.10'
+
+        }
```
Comment (macintosh1398) [inline build.gradle:7]: It's a detail but maybe reformat the code here.

**Review by macintosh1398 — COMMENTED** [review]
You did a good job with Firebase, look promising. The comments of @Matathiasq13824 are very relevant. Mine are more based on details, put it's always good to have a good formatting (and adding some comments), especially in the future when we'll have a lot of code.

File: app/src/main/java/com/github/multimatum_team/multimatum/repository/DeadlineRepository.kt (hunk 1 above)
Comment (felko) [reply app/src/main/java/com/github/multimatum_team/multimatum/repository/DeadlineRepository.kt]: Yes, it is meant to be defined in implementing classes.
`FirebaseDeadlineRepository` will implement this function by storing the provided deadline in Firebase.
I decided to use an interface so I can implement a mock `DeadlineRepository` for testing purposes.

**Comment by macintosh1398** [comment]
Looks good to me, the commented sections were corrected. Resolve the merging and testing problems and it is ok for me.

**Review by macintosh1398 — APPROVED** [review]
You got my approval. Just solve the current issues.

**Review by Matathiasq13824 — CHANGES_REQUESTED** [review]
Same as @macintosh1398, correct the tests and make the merge and everything should be good

**Review by Matathiasq13824 — APPROVED** [review]
As the tests works but it is the calculated coverage that fails the CI, I don't see anymore problem

**Pull request merged**